terraform/modules/kubernetes/kubernetes.tf

```diff
@@ -205,15 +201,48 @@ resource "kubernetes_cron_job" "workflow_manager" {
             # "Never".
             # https://kubernetes.io/docs/concepts/workloads/controllers/job/#handling-pod-and-container-failures
             # https://github.com/kubernetes/kubernetes/issues/74848
-            restart_policy       = "Never"
-            service_account_name = kubernetes_service_account.workflow_manager.metadata[0].name
+            restart_policy                  = "Never"
+            service_account_name            = kubernetes_service_account.workflow_manager.metadata[0].name
+            automount_service_account_token = true
           }
         }
       }
     }
   }
 }
 
+resource "kubernetes_role" "workflow-manager-role" {
+  metadata {
+    name      = "${var.environment}-${var.data_share_processor_name}-wfm-role"
+    namespace = var.kubernetes_namespace
+  }
+
+  rule {
+    api_groups = [""]
+    resources  = ["namespaces", "pods"]
+    verbs      = ["get", "list", "watch"]
+  }
+}
+
+resource "kubernetes_role_binding" "workflow-manager-rolebinding" {
```

Comment:
nit: I like to use snake_case labels (i.e., `workflow_manager_rolebinding` so that then when you're referencing this object elsewhere, you get a consistent dotted identifier like `kubernetes_role_binding.workflow_manager_rolebinding`. However `terraform fmt` has no opinions about this and I'm sure I have some kebab case somewhere in here.